### DISCUSSION TITLE
Rule proposal: template/prefer-control-flow-syntax

### DISCUSSION THREAD

**Discussion by the-ult** [angular-eslint/angular-eslint#1592] [New Rule Proposals]
With the release of Angular 17 and the new [Control Flow Syntax](https://angular.dev/guide/templates/control-flow), it would be great if there would be a new (recommended) rule, that checks whether the new Control Flow Syntax is used instead of the current/old directives (`NgIf`, `NgFor`, NgSwitch`).

When using Angular 17, this could be part of `recommended` and maybe even have an auto-fix.

see: https://github.com/angular-eslint/angular-eslint/issues/1591

**Comment by json-derulo** [comment]
I think it's too early to be recommended. Control Flow is in developer preview, so it can have breaking changes. It's not ready for productive use yet, the IDEs and tooling need to adjust first. Also, the template parser possibly needs to be adjusted to support the new syntax. 

Additionally, I don't think that an auto-fix should be provided, as Angular itself will provide schematics to migrate from the directives to the control flow.

**Reply by JamesHenry** [reply]
Few points:

- The template-parser already supports the syntax as of v17: https://github.com/angular-eslint/angular-eslint/releases/tag/v17.0.0
- I agree it's not ready to be in `recommended` yet, we can potentially look for v18 if the someone has made the rule by then
- I agree that the auto-fix would be duplicating the work of the Angular team with their schematics
- I would accept a PR for such a rule as I think it's clear that's where the framework is headed

**Reply by json-derulo** [reply]
The `@angular-eslint/template/prefer-control-flow` rule is available since v17.1.0 🚀

**Reply by SchroederSteffen** [reply]
Any plans to add this to `recommended` in `v21`? The old syntax is intended to be removed with `v22`: https://angular.dev/api/common/NgIf

**Reply by JamesHenry** [reply]
Thanks @SchroederSteffen, tracked here: https://github.com/angular-eslint/angular-eslint/issues/2614
